src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -191,7 +194,7 @@ export default class DefaultMessagingSession implements MessagingSession {
       !this.isClosing &&
       this.canReconnect(event.code) &&
       this.reconnectController.retryWithBackoff(async () => {
-        this.startConnecting(true);
+        await this.startConnecting(true);
```

Comment:
`await`ing this should also not affect as for the successful connection we will be depending on the `messagingSesssionDidStart` right? Tagging Kyu (@simmkyu ) for checking once.

Reply:
Right, `await` is unnecessary because `this.isConnecting` will be `true` anyway.

Reply:
But the `startConnecting` actually does not check for `isConnecting` may be instead of doing the check for `isConnecting` in `start` would it be better to do it in `startConnecting`?

Reply:
oh I understand what you meant. For example, if the WebSocket receives the `close` event twice in a short period, the messaging session can run `startConnecting` twice.

If we check for `this.isConnecting` in the `startConnecting`, we can guarantee that the messaging session executes `startConnecting` once at a time.

Reply:
@dpwspoon seems like to avoid running into duplicate reconnects, moving the `isConnecting` check inside the `startConnecting` will help over using `await`. Also, if waiting for reconnection means an end to end success, should we set the `isConnecting` to `false` in the `messagingSessionDidStart`?

Reply:
Plus didStart might never trigger say in auth failure